### ISSUE TITLE
Major bug in ACCEL's algorithm for selecting easy edit levels 

### ISSUE THREAD

**Issue by JaydenTeoh** [facebookresearch/dcd#11]
@minqi Hi, my team and I have been working on a research to improve upon ACCEL for a while now. We were unable to reproduce the results shown in the original paper and I suspect it is due to a bug in the code for ACCEL's edit level selection. Please see the lines below:

https://github.com/facebookresearch/dcd/blob/81c9f36c8c72f51ed1640b8227aaf3b812e5129b/envs/runners/adversarial_runner.py#L765
https://github.com/facebookresearch/dcd/blob/81c9f36c8c72f51ed1640b8227aaf3b812e5129b/envs/runners/adversarial_runner.py#L769

Note that the regret numpy array `agent_info['mean_return'].detach().cpu().numpy() - agent_info['batched_value_loss'].detach().cpu().numpy()` is a 2D-array
<img width="181" alt="image" src="https://github.com/facebookresearch/dcd/assets/90945854/048fe425-3354-466d-9df3-a3adccf2612c">

which means the easy-indices,`easy = list(np.argsort((agent_info['mean_return'].detach().cpu().numpy() - agent_info['batched_value_loss'].detach().cpu().numpy()))[:4])`, returns `[0, 0, 0, 0]` because the regret numpy array is not flattened

<img width="349" alt="image" src="https://github.com/facebookresearch/dcd/assets/90945854/72dfef97-2258-47db-9186-67a6e1d61e05">

which results in ACCEL only editing the first-indexed seed in the replay seeds.

I've made a PR with the fix to the bug and will reference this issue in the PR.

---------------------------------------------------------------------------------
On this note , may I confirm that this bug is a part of the original ACCEL algorithm and whether it affects the results for the easy selection ACCEL? We also noticed there is a significant difference in the results for ACCEL for Minigrid-60 comparing the original paper and CoLLAs 2023.

ACCEL:
<img width="82" alt="image" src="https://github.com/facebookresearch/dcd/assets/90945854/18650e29-28e8-4f21-ae4f-2884c1e0e037">

CoLLAs 2023:
<img width="77" alt="image" src="https://github.com/facebookresearch/dcd/assets/90945854/99bf05ae-ede4-4c7a-9e37-14995ddbba61">

Thank you and I look forward to your response.

**Comment by minqi** [comment]
Hi @JaydenTeoh, we're excited to hear you are working to improve ACCEL. Thanks for spotting this bug! Indeed, this means we are simply generating mutations using a single random level in the batch during the mutation step. We will update the arXiv version of ACCEL to reflect this finding.

The results from the paper were produced using the same code. However, the difference is that during evaluation in the original ACCEL paper, 50 parallel processes were used, which introduced an upward bias in the results. The CoLLAs do not do this, so those evaluation scores should be used for comparison (see Appendix B in the CoLLAs paper for details on this difference).

**Comment by JaydenTeoh** [comment]
@minqi Appreciate the prompt response! Thank you for the amazing work on UED.
